Cython/Compiler/ExprNodes.py

```diff
@@ -9265,6 +9267,97 @@ def generate_operation_code(self, code):
             raise InternalError("List type never specified")
 
 
+class CArrayNode(ListNode):
+    """A homogeneous C array of C values.
+
+    Created through coercion from ListNode.
+    """
+
+    is_temp = False
+    array_cname = None
+    _array_cname_temp = None
+
+    def _non_const_array_type(self):
+        array_type = self.type
+        if array_type.base_type.is_const:
+            array_type = PyrexTypes.c_array_type(
+                PyrexTypes.remove_cv_ref(array_type.base_type), array_type.size)
+        return array_type
+
+    def _allocate_carray(self, code, as_const):
+        array_type = self.type
+        if as_const:
+            # Declare static const array in place.
+            self.array_cname = code.globalstate.new_const_cname('carray')
```

Comment:
I don't think it does any harm, but all the examples I can find are allocated in a local scope (either the module-init function, or in an individual function, or in the closure struct) so the names don't need to be globally unique.

Although maybe it makes them easier to reason about if they are.